### PR TITLE
[CSM-160] Return HTTP 400 Bad Request for malformed requests

### PR DESCRIPTION
**Pull request by Pastafarianist**

### CODE REVIEW

File: src/Pos/Explorer/Web/ClientTypes.hs [hunk 1]
```diff
@@ -87,14 +87,15 @@ encodeHashHex :: Hash a -> Text
...
-decodeHashHex = fmap Bi.decode . processRes . B16.decode . encodeUtf8
-  where processRes (res, rest) =
-            if BS.null rest
-            then Right $ BSL.fromStrict res
-            else Left $ "decodeHashHex: couldn't decode rest of hash: " <> decodeUtf8 rest
-
-decodeHashHex' :: Text -> Hash a
-decodeHashHex' = either (error "decodeHashHex: invalid hash") identity . decodeHashHex
+decodeHashHex hashText = do
+    hashBinary <- processRes . B16.decode . encodeUtf8 $ hashText
```
Comment (Martoon-00) [inline src/Pos/Explorer/Web/ClientTypes.hs]: You can make your life easier if use `Serokell.Util.Base16.decode`:
https://github.com/serokell/serokell-util/blob/master/src/Serokell/Util/Base16.hs#L22

File: src/Pos/Explorer/Web/Server.hs [hunk 2]
```diff
@@ -179,12 +179,14 @@ getLastTxs (fromIntegral -> lim) (fromIntegral -> off) = do
     pure $ [toTxEntry (tiTimestamp txi) (tiTx txi) | txi <- localTxsWithTs <> blockTxsWithTs]
 
 getBlockSummary :: ExplorerMode m => CHash -> m CBlockSummary
-getBlockSummary (fromCHash' -> h) = do
+getBlockSummary cHash = do
+    h <- either (throwM . Internal) pure $ fromCHash cHash
```
Comment (flyingleafe) [inline src/Pos/Explorer/Web/Server.hs]: It's probably better to make it separate helper (like `maybeThrow`). But it's cosmetics

File: src/Pos/Explorer/Web/ClientTypes.hs (hunk 1 above)
Comment (Pastafarianist) [inline src/Pos/Explorer/Web/ClientTypes.hs]: Fixed.

File: src/Pos/Explorer/Web/Server.hs (hunk 2 above)
Comment (Pastafarianist) [inline src/Pos/Explorer/Web/Server.hs]: Fixed (`unwrapOrThrow`).

**Pull request merged**